lib/redlock/client.rb

```diff
@@ -203,8 +203,13 @@ def lock_instances(resource, ttl, options)
       value = (options[:extend] || { value: SecureRandom.uuid })[:value]
       allow_new_lock = options[:extend_only_if_locked] ? 'no' : 'yes'
 
-      locked, time_elapsed = timed do
-        @servers.select { |s| s.lock resource, value, ttl, allow_new_lock }.size
+      locked = 0
+      time_elapsed = timed do
+        @servers.each do |server|
+          if server.lock(resource, value, ttl, allow_new_lock)
+            locked += 1
```

Comment:
TBH, I don't think this easier to read, especially mutating `locked` in an conditional.

If it's only about the `timed` function having 2 return values, you could still do

```rb
locked = 0
time_elapsed = timed do
  locked = @servers.select { |s| s.lock resource, value, ttl, allow_new_lock }.size
end
```

but on the other hand, maybe it's only me, and I wouldn't say I had a strong opinion on this one :) So if you insist, please go ahead.

Reply:
Ya, one of the main reasons was 2 return values. Other, not critical in this case, is avoiding extra array allocation.
While original code is still very simple to understand it fast, personally I find it more clear when `.select`-like methods are using obvious explicit boolean-like methods inside it, like `.locked?` for example, to filter collection. And for *doing something on* collection, I would prefer just iterating using `.each`.

Reply:
hmm, so only `.each` may have side-effects? That's quite confining, isn't it?

How about

```rb
@servers.map { |s| s.lock resource, value, ttl, allow_new_lock }
                .select(&:itself)   # Or .select { |locked| locked }
                .size
```

wrt. to the extra array allocation: Seems a bit too detailed for me, without having any benchmarks/measurements.

But again, I don't have any strong opinions on any of this. Maybe @leandromoreira, but otherwise we can also just merge this in, from my pov.

Reply:
FWIW There's is a `.count` method on arrays:

```ruby
@servers.count{ |s| s.lock resource, value, ttl, allow_new_lock }
```

Reply:
@maltoe Agreed with your initial comment, so updated taking it into account.

Reply:
@kitsunde Yes, but `[1, 2, nil].count == 3`

Reply:
When block is provided, `count` returns elements for which block evaluates to truthy value. So this suggestion is good here.

Reply:
Oh nice, today I learned 👍